java/maven/src/org/netbeans/modules/maven/problems/MavenModelProblemsProvider.java

```diff
@@ -179,25 +177,50 @@ private Pair<Collection<ProjectProblem>, Boolean> doGetProblems1(boolean sync) {
             
             }
             MavenProject o = analysedProject.get();
-            LOG.log(Level.FINER, "Called getProblems for {0}, analysed = {1}, current = {2}", 
-                    new Object[] { project, o == null ? 0 : System.identityHashCode(o), System.identityHashCode(updatedPrj) });
             //for non changed project models, no need to recalculate, always return the cached value
-            Object wasprocessed = updatedPrj.getContextValue(MavenModelProblemsProvider.class.getName());
-            if (o == updatedPrj && wasprocessed != null) {
-                Pair<Collection<ProjectProblem>, Boolean> cached = problemsCache;
-                LOG.log(Level.FINER, "getProblems: Project was processed, cached is: {0}", cached);
-                if (cached != null) {
-                    return cached;
+            if (pending.isDone()) {
+                try {
+                    // cannot block, if .isDone().
+                    MavenProject updatedPrj = pending.get();
+                    LOG.log(Level.FINER, "Called getProblems for {0}, analysed = {1}, current = {2}", 
+                            new Object[] { project, o == null ? 0 : System.identityHashCode(o), System.identityHashCode(updatedPrj) });
```

Comment:
could simply use `System.identityHashCode(o)` since it already returns `0` if the argument is null.